### ISSUE TITLE
Database first approach dbContext cannot query data after updated to 2.0.0 . Give 502.3 Bad Gateway error.

### ISSUE THREAD

**Issue by DNS-Cihang** [dotnet/efcore#9486]
I cannot do anything about DbContext after updated to EntityframeworkCore 2.0.0. Even I created a new Aspnt.Core 2.0.0 app, and scaffolding models from existing database, the same error happened.

### Steps to reproduce
* Scaffolding models from existing database.
* Follow the official tutorial to register dbContext in StartUp.cs
`services.AddDbContextPool<SomeDbContext>(
          options => options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));`
* Register dbContext in controller and try query something
![Issue screenshot](http://res.cloudinary.com/dvl0bnhhw/image/upload/v1503196794/2017-08-19_22_39_23-Entity_Debugging_-_Microsoft_Visual_Studio_fzjdao.png)

Try to evaluate the context and got this
![Issue screenshot](http://res.cloudinary.com/dvl0bnhhw/image/upload/v1503196415/2017-08-19_22_21_37-Entity_-_Microsoft_Visual_Studio_cefu95.png)

### Further technical details
EF Core version: 2.0.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer 2.0.0
Operating system: Windows 10
IDE: Visual Studio 2017 15.3.1

**Comment by johanskoldekrans** [comment]
I haven't been able to get this to work either and the documentation on this is lacking. 

If I scaffold using both Efcore 2.0 and 2.1-preview1 and follow the instructions from the official tutorial I get:

`
System.ArgumentException occurred
  HResult=0x80070057
  Message=AddDbContext was called with configuration, but the context type 'SomeContext' only declares a parameterless constructor. This means that the configuration passed to AddDbContext will never be used. If configuration is passed to AddDbContext, then 'SomeContext' should declare a constructor that accepts a DbContextOptions<SomeContext> and must pass it to the base constructor for DbContext.
  Source=<Cannot evaluate the exception source>
  StackTrace:
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.CheckContextConstructors[TContext]()
   at Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.AddDbContextPool[TContext](IServiceCollection serviceCollection, Action`2 optionsAction, Int32 poolSize)
   at Project.Startup.ConfigureServices(IServiceCollection services) in Startup.cs:line 55
`
Further technical details

EF Core version: 2.1.0-preview1-26723
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017 15.3.1

**Comment by seb999** [comment]
Hi Johan,

Perhaps you forgot to add in your context class the following contructor:

`
public class MISContext : DbContext
{
public MISContext(DbContextOptions<MISContext> options) : base(options)
{ }

protected override void OnModelCreating(ModelBuilder modelBuilder)
{
.......
.......
}
`

**Comment by johanskoldekrans** [comment]
I did that about ten times when I started with efcore but every time you scaffold it removes it. Since this is best practice I have a hard time understanding why it is removed every time... Out of the box the solution isnt working so this should be included in the documentation on how to implement the ContextPool.

**Comment by seb999** [comment]
I'm also like you stuck with migration to 2.0, I opened this issue few days ago:
https://github.com/aspnet/EntityFrameworkCore/issues/9462

**Comment by badyalsk** [comment]
I'm also looking into the solution to solve this chaos.  

I had tried:

Scaffold-DbContext in an empty project replaced all Entity classes and Database SomeDbContext class in Project.

Registered dbContext in StartUp.cs
services.AddDbContextPool<SomeDbContext>( options => options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));

Added constructor in SomeDbContext class like
public SomeDbContext (DbContextOptionsoptions) : base(options)
{
}

 but all these steps didn't solve the havoc.
